### PR TITLE
INT B-20879 SIT panel & SIT extension fixes and calculation fixes

### PR DESCRIPTION
**Pull request by danieljordan-caci**
## Summary

When demoing SIT functionality to HomeSafe, it was found that there were some calculations that were a little off in the SIT panel and when reviewing/editing SIT extension dates/days. In SIT calculations, we had two instances where we were subtracting days (Truss did this for some weird reason) and that was causing the SIT panel to say and show inconsistent information when changing things in the SIT modal.

There were also some suggested UI fixes of wording.

This PR does the following:
- added a `sitFormatters` JSX component that contains reusable functions and renders used by `ReviewSITExtensionModal` and `SubmitSITExtensionModal`
- refactored both above components to use the shared utility functions to simplify the logic and help further changes needing to be in once place if they occur
- adjusted calculations to reflect request changes from the gov
- added some stylings that improved the alignment and readability of some components (like those stupid date picker boxes being misaligned)
- updated playwright tests
- added relevant tests

### How to test

1. Access MilMove as an office user
2. Find a move or add origin/destination SIT to a move
3. Approve the SIT items as TOO
4. You should now be able to see the SIT panel data
5. I used [this tool](https://www.timeanddate.com/date/duration.html?y1=2024&m1=7&d1=15&y2=2024&m2=8&d2=6) to calculate days
6. Keep in mind that we want to include the SIT entry date & the SIT departure date in the calculations
7. In the Prime Sim, request a SIT extension for that shipment
8. Go back as the TOO and review the request, change some dates/values around and make sure that nothing is wonky or inaccurate

## Screenshots
SIT panel & Edit SIT authorization should reflect the same data

SIT panel
Screenshot 2024-08-06 at 2 16 09 PM

SIT auth edit (AKA `SubmitSITExtensionModal`
Screenshot 2024-08-06 at 2 16 17 PM

SIT panel & review SIT extension modal should reflect similar data

SIT panel
Screenshot 2024-08-06 at 2 18 43 PM

SIT extension review modal (AKA `ReviewSITExtensionModal`)
Screenshot 2024-08-06 at 2 18 55 PM

Showing that when I change the approved days, the SIT auth end date also changes. Biggest thing here is the `Approved + Requested` date should match up when you put input the max number of days that can be approved.

Also showing that it works backwards.
https://github.com/user-attachments/assets/3ccaa9fc-9858-452a-9ff5-ba5de4a6832e

### CODE REVIEW

File: src/utils/sitFormatters.jsx [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+      mask={Number}
+      lazy={false}
+      scale={0}
+      signed={false}
+      inputClassName={styles.approvedDaysInput}
+      errorClassName={styles.errors}
+      onChange={onChange}
+    />
+  </div>
+);
```
Comment (cameroncaci) [inline src/utils/sitFormatters.jsx:134]: Label of "Total days of SIT approved " is not recognized for 508 compliance
image

**Review by cameroncaci — CHANGES_REQUESTED** [review]
Meets AC, the SIT modal is still confusing but you did a great job with this

File: src/utils/sitFormatters.jsx (hunk 1 above)
Comment (danieljordan-caci) [reply src/utils/sitFormatters.jsx:134]: Fixed

Screenshot 2024-08-09 at 8 32 00 AM

Comment (cameroncaci) [reply src/utils/sitFormatters.jsx]: Hm, now there are two labels for one field
image

**Review by cameroncaci — CHANGES_REQUESTED** [review]
Camelcase label now showing https://github.com/transcom/mymove/pull/13405#discussion_r1711482196

File: src/utils/sitFormatters.jsx (hunk 1 above)
Comment (cameroncaci) [reply src/utils/sitFormatters.jsx]: I believe the best approach would be to somehow get "Total days of SIT approved" to be recognized again as the proper label

Comment (danieljordan-caci) [reply src/utils/sitFormatters.jsx]: Per our convo in Slack... you know what I had to do. I'm not proud of it, but I did it.

**Comment by danieljordan-caci** [comment]
Flaky tests failing

**Review by TevinAdams — APPROVED** [review]
Tested and reviewed. Test case worked and calculations look correct.

**Pull request merged**